src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Automation/RunCodeGeneration.ps1

```diff
@@ -1317,6 +1374,31 @@ function Write-OperationCmdletFile
             $paramTypeNormalizedName = Get-NormalizedTypeName -inputName $paramTypeFullName;
             $param_constructor_code = Get-ConstructorCodeByNormalizedTypeName -inputName $paramTypeNormalizedName;
 
+            $is_special_type = Is-SpecialType -inputName $paramTypeFullName;
+            $has_properties = $true;
+            if ($is_special_type)
+            {
+                $properties_of_special_type = $pt.ParameterType.GetProperties();
+                if ($properties_of_special_type.Count -eq 0)
+                {
+                     $has_properties = $false;
+                }
+            }
+
+            $is_string_list = $null;
+            $does_contain_only_strings = $null;
+            if ($has_properties)
+            {
+                $is_string_list = Is-ListType -property_info_array $pt.ParameterType.GetProperties();
+                $does_contain_only_strings = Does-ContainOnlyStringTypes -property_info_array $pt.ParameterType.GetProperties();
+            }
+
+            $only_strings = $true;
```

Comment:
Please add some comment to introduce the code logic. I think you are trying to search for a type that is: 1) complex model class type, 2) and it's a list of strings?